src/sds_data_model/_raster.py

```diff
@@ -0,0 +1,466 @@
+                expected_x_max=expected_x_max,
+                expected_y_max=expected_y_max,
+                nodata=nodata,
+            )
+
+    _dataset = merge(
+        data_array.to_dataset(name=data_array.name) for data_array in data_arrays
+    )
+    dims = ("eastings", "northings")
+    if set(_dataset.dims) != set(dims):
+        _dataset = _dataset.rename(
+            {dataset.rio.x_dim: dims[0], dataset.rio.y_dim: dims[1]}
+        )
+    _dataset.update(
+        affine_to_coords(
+            Affine(
+                expected_cell_size,
+                0,
+                expected_x_min,
+                0,
+                -expected_cell_size,
+                expected_y_max,
+            ),
+            height=int(expected_y_max / expected_cell_size),
+            width=int(expected_x_max / expected_cell_size),
+            y_dim=dims[1],
+            x_dim=dims[0],
+        ),
+    )
+    _dataset.rio.set_spatial_dims(*dims)
+    return _dataset
+
+
+def _read_dataset_from_file(
+    data_path: str,
+    bands: Optional[Union[List[int], List[str]]] = None,
+    categorical: Union[bool, Dict[Union[int, str], bool]] = False,
+    nodata: Optional[float] = None,
+    engine: Optional[str] = None,
+    decode_coords: Optional[Union[bool, Literal["coordinates", "all"]]] = "all",
+    expected_cell_size: int = CELL_SIZE,
+    expected_x_min: int = BNG_XMIN,
+    expected_x_max: int = BNG_XMAX,
+    expected_y_max: int = BNG_YMAX,
+) -> Dataset:
+    """Returns a `Dataset` from an input raster according to the constant value args.
+
+    Opens the raster from the input filename using `open_dataset`. CRS is checked.
+    Data and args are passed to `_resample_and_reshape` that will conditionally
+    resample and reshape the input data.
+
+    Args:
+        data_path (str): File path to raster.
+        bands (Optional[Union[List[int], List[str]]], optional): List of bands to select
+            from the `Dataset`. Defaults to None.
+        categorical (Union[bool, Dict[Union[int, str], bool]], optional):
+            `bool` or `dict` mapping ({band : bool}) of the interpolation used
+            to resample. Defaults to False.
+        nodata (Optional[float], optional): Value that will fill the grid where
+            there is no data (if it is not `None`). Defaults to None.
+        engine (Optional[str], optional): Engine used by `open_dataset`.
+            Defaults to None.
+        decode_coords (Optional[Union[bool, Literal["coordinates", "all"]]], optional):
+            Value used by `open_dataset`. Variable upon `engine` selection.
+            Defaults to "all".
+        expected_cell_size (int): Cell size to resample the data to if it is not already
+            this value. Defaults to CELL_SIZE.
+        expected_x_min (int): x minimum. Defaults to BNG_XMIN.
+        expected_x_max (int): x maximum (width). Defaults to BNG_XMAX.
+        expected_y_max (int): y maxium (height). Defaults to BNG_YMAX.
```

Comment:
maximum is missing an "m".

```suggestion
        expected_y_max (int): y maximum (height). Defaults to BNG_YMAX.
```